objc/objc_runtime_darwin.go

```diff
@@ -152,17 +155,53 @@ func (c Class) AddMethod(name SEL, imp _IMP, types string) bool {
 	return byte(ret) != 0
 }
 
-// Register registers a class that was allocated using objc_allocateClassPair.
+// AddIvar adds a new instance variable to a class.
+// It may only be called after AllocateClassPair and before Register.
+// Adding an instance variable to an existing class is not supported.
+// The class must not be a metaclass. Adding an instance variable to a metaclass is not supported.
+// The instance variable's minimum alignment in bytes is 1<<align. The minimum alignment of an
+// instance variable depends on the ivar's type and the machine architecture.
+// For variables of any pointer type, pass math.Log2(unsafe.Alignof(type)).
+func (c Class) AddIvar(name string, size uintptr, alignment uint8, types string) bool {
+	n := strings.CString(name)
+	t := strings.CString(types)
+	ret, _, _ := purego.SyscallN(class_addIvar, uintptr(c), uintptr(unsafe.Pointer(n)), size, uintptr(alignment), uintptr(unsafe.Pointer(t)))
+	runtime.KeepAlive(n)
+	runtime.KeepAlive(t)
```

Comment:
Do we need `runtime.KeepAlive` for C pointers? I don't think so. Rather, we have to free them explicitly like

```
n := strings.CString(name)
defer C.free(unsafe.Pointer(n))
```

Reply:
I realized there are similar things in other functions. Please update them. Thanks,

Reply:
Or I'll send a PR for other existing part as Ebitengine already depends on it and this might cause memory leak so the fix is necessary asap...

Reply:
This isn't a C pointer. It's a pointer to Go memory. The function definition is in internal/strings

Reply:
Oh sorry I was misunderstanding.. Thanks!